python/desc/imsim/flats.py

```diff
@@ -0,0 +1,90 @@
+"""
+imSim module to make flats.
+"""
+import galsim
+import desc.imsim
+
+__all__ = ['make_flat']
+
+
+def make_flat(gs_det, counts_per_iter, niter, rng, buf=2, logger=None,
+              wcs=None):
+    """
+    Create a flat by successively adding lower level flat sky images.
+    This is based on
+    https://github.com/GalSim-developers/GalSim/blob/releases/2.0/devel/lsst/treering_flat.py
+
+    Full LSST CCDs are assembled one amp at a time to limit the memory
+    used by the galsim.SiliconSensor model.
+
+    Parameters
+    ----------
+    gs_det: GalSimDetector
+        The detector in the LSST focalplane to use.  This object
+        contains the CCD pixel geometry, WCS, and treering info.
+    counts_per_iter: int
+        Roughly the number of electrons/pixel to add at each iteration.
```

Comment:
"Roughly" reads funny, like there is some uncertainty in how much will be added.  And while that's true for each pixel, the mean value is fairly precisely equal to `counts_per_iter`.  So I'd suggest writing "Mean number of electrons/pixel" instead to be clearer.